swupd/manifest.go

```diff
@@ -342,72 +342,113 @@ func (m *Manifest) sortFilesVersionName() {
 	})
 }
 
-func (m *Manifest) addDeleted(oldManifest *Manifest) {
-	for _, df := range oldManifest.DeletedFiles {
-		if df.findFileNameInSlice(m.Files) == nil {
-			m.Files = append(m.Files, df)
-			m.DeletedFiles = append(m.DeletedFiles, df)
-		}
-	}
+func (m *Manifest) appendFile(f *File) {
```

Comment:
Optional: unless there is a reason to have (for future changes), I would remove this method.

Reply:
+1